DataGenerator.py

```diff
@@ -0,0 +1,221 @@
+        if type is KindOfData.LIVE:
+            n = len(cls.data_live)
+            return cls.data_live[int(interval[0] * n): int(interval[1] * n)], cls.mask_live[int(interval[0] * n): int(interval[1] * n)]
+        if type is KindOfData.ALL:
+            n = len(cls.data_audioset)
+            m = len(cls.data_live)
+
+            # cut some piece of data
+            data_a = cls.data_audioset[int(interval[0] * n): int(interval[1] * n)]
+            mask_a = cls.mask_audioset[int(interval[0] * n): int(interval[1] * n)]
+            data_l = cls.data_live[int(interval[0] * m): int(interval[1] * m)]
+            mask_l = cls.mask_live[int(interval[0] * m): int(interval[1] * m)]
+            data = np.vstack([data_a, data_l])
+            mask = np.vstack([mask_a, mask_l])
+
+            # shuffle embeddings and masks of different sets in the same order
+            order = np.random.permutation(mask.shape[0])
+            data = cls.__shuffle(data, order)
+            mask = cls.__shuffle(mask, order)
+            return data, mask
+
+    @classmethod
+    def generate_audioset_sample(cls, n_samples, path_to_data='bal_train.h5',
+                                 path_to_labels='class_labels_indices.csv',
+                                 path_to_genres=None):
+        """Generate n samples using AudioSet"""
+
+        # extract info from audio set
+        music_data, speech_data = cls.__extract_info_audioset(path_to_data, path_to_labels)
+
+        # extract info about music with genres classification
+        if path_to_genres:
+            music_data_new_res = cls.__extract_music_for_genres(path_to_genres)
+            # add data to sample
+            music_data = np.vstack([music_data, music_data_new_res])
+
+        # generate samples
+        cls.data_audioset, cls.mask_audioset = cls.__generate_data_sample(music_data, speech_data, n_samples)
+
+    @classmethod
+    def generate_live_sample(cls, path):
+        """Generate samples from YouTube video, films etc"""
+
+        # read data
+        with open(path, "rb") as handle:
+            data_dict = pickle.load(handle)
+        live_embed = cls.__uint8_to_float32([data_dict['embeddings_list']][0])
+        mask_list = cls.__bool_to_float32([data_dict['mask_list']][0])
+
+        # shuffle embeddings and masks in the same order
+        order = np.random.permutation(mask_list.shape[0])
+        cls.data_live = cls.__shuffle(live_embed, order)
+        cls.mask_live = cls.__shuffle(mask_list, order)
+        # np.zeros((N_samples, seq_len, 1), dtype=np.float32)
+        sh = cls.mask_live.shape
+        cls.mask_live = cls.mask_live.reshape(sh[0], sh[1], 1)
+
+
+    @staticmethod
+    def __shuffle(arr, order):
+        new_arr = np.zeros(arr.shape)
+        for i in range(len(order)):
+            new_arr[order[i]] = arr[i]
+        return new_arr
+
+    @staticmethod
+    def __generate_data_sample(music_data, speech_data,
```

Comment:
Here's docstring comment for the legacy version of this function, it may be found useful to write docstring for the function:
`     """Generates array of data samples for training or testing model having array of music and speech samples

    Args:
        music_data: complete numpy array of shape (len, dur, emb), where len is the number of music pieces 
        with the duration of dur seconds each with emb-dimensional vector of embeddings
        
        speech_data: the same as music_data but sound pieces correspond "is_speaking" class
        
        N_music_min: index of the first sound piece to be used (in music class)
        
        N_music_max: index of the last sound piece to be used (in music class)
        
        N_speech_min: index of the first sound piece to be used (in speech class)
        
        N_speech_max: index of the last sound piece to be used (in speech class)
        
        N_samples: the number of samples to generate
        
        seq_len: sequence length in seconds, the max length of music part in a single sample
        
        embed_dim: the dimension of embeddings (the third dim of music_data shape)
        
    Returns:
        (data_sample, mask_sample) where
        data_sample: numpy tensor of samples with speech and music parts, shape is (sample_index, time, embeddings)
        
        mask_sample: numpy tensor with masks of the samples with ones in place of music part
    """ `

Reply:
Thx, Nikita)